packages/kbn-openapi-generator/src/lib/get_generated_file_path.ts

```diff
@@ -7,5 +7,7 @@
  */
 
 export function getGeneratedFilePath(sourcePath: string) {
-  return sourcePath.replace(/\..+$/, '.gen.ts');
+  // Remove any double extension like `.schema.yaml` or `.schema.yml` and replace with `.gen.ts`
+  const secondToLastDot = sourcePath.lastIndexOf('.', sourcePath.lastIndexOf('.') - 1);
+  return `${sourcePath.substring(0, secondToLastDot)}.gen.ts`;
```

Comment:
Thanks for fixing the `firstname.` [issue](https://github.com/elastic/kibana/pull/174317#discussion_r1448109094) for some developers when running `yarn openapi:generate` 🙏